lusid-sdk-python/tests/finbournetest.py

```diff
@@ -431,6 +442,154 @@ def create_property_definition_test(self, property_request, property):
+                                                 effective_at=effective_at)
+            # Convert to a dictionary with the instrument uid as the key and a list of holdings as values
+            holdings = {}
+            for holding in holdings_.values:
+                holdings.setdefault(holding.instrument_uid, []).append(holding)
+            return holdings
+
+        def aggregate_holdings(holdings):
+            """
+            This function takes a list of holdings for a given instrument and if there is more than one aggregates the
+            number of units and the cost of the holdings.
+
+            :param holdings: A list of holdings for a given instrument
+
+            :return:
+            holding_units: The total number of units held in this instrument
+            holding_cost_amount: The total cost of units held in this instrument
+            holding_cost_currency: The currency of units held in this instrument
+            """
+
+            # If we have more than one holding, use our aggregation logic
+            if len(holdings) > 1:
+                # Extract our units from each of our holdings objects
+                holding_units_ = [holding.units for holding in holdings]
+                # Extract our cost amount from each of our holdings objects
+                holding_cost_amount_ = [holding.cost.amount for holding in holdings]
+                # Sum the units and cost amount, rounding to the appropriate level of decimal places
+                holding_units = round(reduce((lambda x, y: x + y), holding_units_), 0)
+                holding_cost_amount = round(reduce((lambda x, y: x + y), holding_cost_amount_), 2)
+
+            # Otherwise just use the first and only holding
+            else:
+                holding_units = round(holdings[0].units, 0)
+                holding_cost_amount = round(holdings[0].cost.amount, 2)
+
+            # Currency is assumed to be the same across all holdings for each instrument
+            holding_cost_currency = holdings[0].cost.currency
+
+            return holding_units, holding_cost_amount, holding_cost_currency
+
+        # Get our left and right holdings
+        left_holdings = get_holdings(left_scope, left_code, left_effective_at)
+        right_holdings = get_holdings(right_scope, right_code, right_effective_at)
+
+        # Convert our reconciliation breaks to a dictionary with the instrument uid as the key
+        rec_breaks = {rec_break.instrument_uid: rec_break for rec_break in reconciliation_break_response.values}
+
+        # Iterate over our left holdings and check that any breaks are correct
+        for key, left_holding in left_holdings.items():
+            # If there is more than one holding for an instrument, aggregate the holdings
+            left_holding_units, left_holding_cost_amount, left_holding_cost_currency = aggregate_holdings(left_holding)
+            # Try and find a holding on the right side for the same instrument
+            try:
```

Comment:
can this be replaced with an ` if ... in ...` rather than using exceptions for control logic

Reply:
Done